src/Microsoft.AspNetCore.JsonPatch/Helpers/CaseInsensitiveDictionary.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.JsonPatch.Exceptions;
+
+namespace Microsoft.AspNetCore.JsonPatch.Helpers
+{
+    internal class CaseInsensitiveDictionary<TKey, TValue> : ICaseInsensitiveDictionary
```

Comment:
How is this different than `new Dictionary<>(StringComparer.OrdinalIgnoreCase)`?